components/src/main/java/com/mercadolibre/android/andesui/button/size/AndesButtonSizeInterface.kt

```diff
@@ -131,70 +136,88 @@ internal interface AndesButtonSizeInterface {
             }
             rightIcon != null -> {
-                handleRightIcon(context, rightIcon, hierarchy)
+                handleStringIcon(context, rightIcon, hierarchy, AndesButtonIconOrientation.RIGHT)
+            }
+            leftDrawable != null -> { // Ignoring if rightIcon is also non null: Left icon has higher precedence than right
+                handleIcon(context, leftDrawable, hierarchy, AndesButtonIconOrientation.LEFT)
+            }
+            rightDrawable != null -> { // Ignoring if rightIcon is also non null: Left icon has higher precedence than right
+                handleIcon(context, rightDrawable, hierarchy, AndesButtonIconOrientation.RIGHT)
             }
             else -> {
                 null // No icon has been specified
             }
         }
     }
 
-    private fun handleLeftIcon(context: Context, leftIcon: String, hierarchy: AndesButtonHierarchyInterface): IconConfig? {
+    private fun handleIcon(context: Context, icon: Drawable,
+        hierarchy: AndesButtonHierarchyInterface, orientation: AndesButtonIconOrientation): IconConfig? {
         return try {
-            val leftBitmapDrawable = buildColoredAndesBitmapDrawable(
-                IconProvider(context).loadIcon(leftIcon) as BitmapDrawable,
+            val bitmapDrawable = buildColoredAndesBitmapDrawable(
+                icon as BitmapDrawable,
                 context,
                 context.resources.getDimensionPixelSize(R.dimen.andes_button_icon_width),
                 context.resources.getDimensionPixelSize(R.dimen.andes_button_icon_height),
                 hierarchy.iconColor(context)
             )
-            IconConfig(leftIcon = leftBitmapDrawable, rightIcon = null)
+            when (orientation) {
+                AndesButtonIconOrientation.LEFT -> IconConfig(leftIcon = bitmapDrawable, rightIcon = null)
+                AndesButtonIconOrientation.RIGHT -> IconConfig(leftIcon = null, rightIcon = bitmapDrawable)
+
+            }
         } catch (e: FileNotFoundException) {
             IconConfig(leftIcon = null, rightIcon = null)
         }
     }
 
-    private fun handleRightIcon(context: Context, rightIcon: String, hierarchy: AndesButtonHierarchyInterface): IconConfig? {
-        return try {
-            val rightBitmapDrawable = buildColoredAndesBitmapDrawable(
-                IconProvider(context).loadIcon(rightIcon) as BitmapDrawable,
-                context,
-                context.resources.getDimensionPixelSize(R.dimen.andes_button_icon_width),
-                context.resources.getDimensionPixelSize(R.dimen.andes_button_icon_height),
-                hierarchy.iconColor(context)
-            )
-            IconConfig(leftIcon = null, rightIcon = rightBitmapDrawable)
-        } catch (e: FileNotFoundException) {
-            IconConfig(leftIcon = null, rightIcon = null)
+    private fun handleStringIcon(context: Context, leftIcon: String,
```

Comment:
Please change this name, because is not a String icon, is the path of the icon.
```suggestion
    private fun handlePathIcon(context: Context, iconPath: String,
```